ethereum-side/contracts/tzNFT.sol

```diff
@@ -10,23 +10,80 @@ contract tzNFT is ERC721URIStorage, Ownable {
     using Counters for Counters.Counter;
     Counters.Counter private _tokenIds;
 
+    mapping (address => mapping (uint256 => uint256)) internalIdsByAddress;
```

Comment:
What is the last `uint256` ?

Reply:
I think that this structure from Openzeppelin could be interesting for us here 
https://docs.openzeppelin.com/contracts/3.x/api/utils#EnumerableSet

`mapping (address => InternalIdSet)`

Reply:
It would simplify `removeTokenOwnership`

Reply:
> What is the last `uint256` ?

The inside mapping is a sort of array with uint256 indexes (incremented by 1 from 0) and tokenIds (uint256 that you were asking for), to be able to iterate easily and use less gas.

Reply:
> I think that this structure from Openzeppelin could be interesting for us here
> https://docs.openzeppelin.com/contracts/3.x/api/utils#EnumerableSet
> 
> `mapping (address => InternalIdSet)`
Top idea, I like this approach. I'll implement it;)